libbpfgo.h

```diff
@@ -131,6 +131,47 @@ int bpf_prog_detach_cgroup_legacy(
   return syscall(__NR_bpf, BPF_PROG_DETACH, &attr, sizeof(attr));
 }
 
+// https://lore.kernel.org/all/20220824233117.1312810-2-haoluo@google.com/
+#ifndef bpf_cgroup_iter_order
+enum bpf_cgroup_iter_order {
+  BPF_ITER_ORDER_UNSPEC = 0,
+  BPF_ITER_SELF_ONLY,        /* process only a single object. */
+  BPF_ITER_DESCENDANTS_PRE,  /* walk descendants in pre-order. */
+  BPF_ITER_DESCENDANTS_POST, /* walk descendants in post-order. */
+  BPF_ITER_ANCESTORS_UP,     /* walk ancestors upward. */
+};
+#define BPF_ITER_LINK_INFO_NO_CGROUP
+#endif
+
+// https://lore.kernel.org/bpf/20220926184957.208194-2-kuifeng@fb.com/
+#ifndef bpf_iter_task_type
+#define BPF_ITER_LINK_INFO_NO_TASK
+#endif
+
+struct bpf_link *bpf_prog_attach_iter(struct bpf_program *prog, __u32 map_fd,
+                                      enum bpf_cgroup_iter_order order,
+                                      __u32 cgroup_fd, __u64 cgroup_id,
+                                      __u32 tid, __u32 pid, __u32 pid_fd) {
```

Comment:
I suppose if we are wrapping `bpf_program__attach_iter()` using upstream `#defines` we should check against `libbpf_major_version()` and `libbpf_minor_version()`. So, in case we don't have these `#defines` and API support, we just declare empty wrappers (or wrappers throwing an error - due to lack of support).

@mozillazg WDYT?

@rafaeldtinoco we need to hear from you as well.